ignite/distributed/auto.py

```diff
@@ -0,0 +1,101 @@
+import sys
+import functools
+import torch
+
+from torch.utils.data import DataLoader
+from torch.utils.data.distributed import DistributedSampler
+
+from ignite.distributed.utils import get_world_size, get_backend, get_local_rank, get_rank, get_device
+from ignite.distributed.utils import initialize, finalize
+
+
+def auto_batch_size(batch_size: int):
+    return batch_size // get_world_size()
+
+
+def auto_num_workers(num_workers: int):
+    if get_backend() == "nccl":
```

Comment:
Something similar should be done for "gloo" and "tpu" ?

Reply:
Yes it should be

Reply:
Done using fresh new `nproc_per_node()` :)